plugins/chat/assets/javascripts/discourse/services/chat-pane-base-subscriptions-manager.js

```diff
@@ -231,6 +231,16 @@ export default class ChatPaneBaseSubscriptionsManager extends Service {
           stagedThread.id = data.thread_id;
           stagedThread.originalMessage.thread = stagedThread;
           stagedThread.originalMessage.thread.preview.replyCount ??= 1;
+
+          // We have to do this because the thread manager cache is keyed by
+          // staged_thread_id, but the thread_id is what we want to use to
+          // look up the thread, otherwise calls to .find() will not return
+          // the thread by its actual ID, and we will end up with double-ups
+          // in places like the thread list when .add() is called.
+          this.model.threadsManager.remove({ id: data.staged_thread_id });
```

Comment:
Seems kind of annoying to do this, but I'm not sure how else to do it. Having some API that replaces one cache key with another seems weird too.

Reply:
Yes Im fine with this for now as long as it's tested and we can iterate :+1: